File: progradio_api/lib/progradio_api/schema/radio.ex
defmodule ProgRadioApi.Radio do
  use Ecto.Schema
  alias ProgRadioApi.{SubRadio, RadioStream, ScheduleEntry, Category, Collection}

  schema "radio" do
    field(:code_name, :string)
    field(:name, :string)
    field(:active, :boolean)
    field(:share, :decimal)
    field(:country_code, :string)
    field(:has_preroll, :boolean)
    has_many(:schedule_entry, ScheduleEntry)
    has_many(:sub_radio, SubRadio)
    has_many(:radio_stream, RadioStream)
    belongs_to(:category, Category)
    belongs_to(:collection, Collection)
  end
end

File: public/vue/types/radio.ts
import type { SubRadio } from './sub_radio';
import type { RadioStream } from './radio_stream';

export interface Radio {
  category: string
  code_name: string
  country_code: string
  id: number
  name: string
  share: number
  streaming_enabled: boolean
  streams: Record<string, RadioStream>
  sub_radios: Record<string, SubRadio>
  type: 'radio'
  img: string
  stream_url: string,
  has_preroll: boolean
}
